### ISSUE TITLE
How to change the text color of code area?

### ISSUE THREAD

**Issue by AleKrabbe** [FXMisc/RichTextFX#737]
I want to implement a dracula theme for the code area, I was able to change the CSS provided for keywords, strings, etc... But the main text in the code area still black. How can I define my own color for it? Thank you.

**Comment by JordanMartinez** [comment]
Have you seen the [CSS Ref Guide](https://github.com/FXMisc/RichTextFX/wiki/RichTextFX-CSS-Reference-Guide)?

**Comment by AleKrabbe** [comment]
Thank you for the reply, I just read it and added 

`.styled-text-area .text {
    -fx-fill: #A9B7C6;
}`

to my css file, but now all the text has changed colors, and I don't have the java keywords highlight anymore. I would like to change the color of just the regular text, not keywords. Here is my whole CSS:

`.styled-text-area .text {
    -fx-fill: #A9B7C6;
}

.keyword {
    -fx-fill: #CC7832;
    -fx-font-weight: bold;
}
.semicolon {
    -fx-fill: #A9B7C6;
    -fx-font-weight: bold;
}
.paren {
    -fx-fill: #A9B7C6;
    -fx-font-weight: bold;
}
.bracket {
    -fx-fill: #A9B7C6;
    -fx-font-weight: bold;
}
.brace {
    -fx-fill: #A9B7C6;
    -fx-font-weight: bold;
}
.string {
    -fx-fill: #6A8759;
}

.comment {
    -fx-fill: #2B2B2B;
}

.paragraph-box:has-caret {
    -fx-background-color: #323232;
}`

**Comment by JordanMartinez** [comment]
Could you provide a reproducible demo?

Also, I think when you need to remove the `text` style class from a text when you are styling it to another style class (e.g. keyword).

**Comment by AleKrabbe** [comment]
How would I do that? I followed the guide provided here: [JavaKeywordsDemo.java](https://github.com/FXMisc/RichTextFX/blob/master/richtextfx-demos/src/main/java/org/fxmisc/richtext/demo/JavaKeywordsDemo.java)

Here is my code:

```
@Override
    public void start(Stage stage) throws Exception {
        Parent root = FXMLLoader.load(getClass().getResource("/fxml/Scene.fxml"));
        
        Scene scene = new Scene(root);
        scene.getStylesheets().addAll("/styles/Styles.css", "/styles/java-keywords.css");
        
        stage.setTitle("Control Flow Grapher - CFG (UFMS)");
        stage.setScene(scene);
        stage.show();
    }

```
```
@Override
    public void initialize(URL url, ResourceBundle rb) {
        CodeArea codeArea = new CodeArea();
        
        // add line numbers to the left of area
        codeArea.setParagraphGraphicFactory(LineNumberFactory.get(codeArea));
        
        // recompute the syntax highlighting 500 ms after user stops editing area
        Subscription cleanupWhenNoLongerNeedIt = codeArea

                // plain changes = ignore style changes that are emitted when syntax highlighting is reapplied
                // multi plain changes = save computation by not rerunning the code multiple times
                //   when making multiple changes (e.g. renaming a method at multiple parts in file)
                .multiPlainChanges()

                // do not emit an event until 500 ms have passed since the last emission of previous stream
                .successionEnds(Duration.ofMillis(500))

                // run the following code block when previous stream emits an event
                .subscribe(ignore -> codeArea.setStyleSpans(0, computeHighlighting(codeArea.getText())));

        // when no longer need syntax highlighting and wish to clean up memory leaks
        // run: `cleanupWhenNoLongerNeedIt.unsubscribe();`
        
        codeArea.replaceText(0, 0, sampleCode);

        stackPane.getChildren().add(new VirtualizedScrollPane<>(codeArea));
        codeArea.setBackground(new Background(new BackgroundFill(Paint.valueOf("#2B2B2B"), CornerRadii.EMPTY, Insets.EMPTY)));

    }
```

The CSS is the one I posted above.

Thank you.

**Comment by JordanMartinez** [comment]
> I ended up just adding the !important tag after setting the text color

I didn't even know about that tag... Thanks for posting!

> Also, I think when you need to remove the text style class from a text when you are styling it to another style class (e.g. keyword).

I believe the `Text` (or perhaps `TextExt`?) class adds the `text` style class to itself. So, even though a portion of the text in the document isn't styled with a `text` style class, it still gets applied.

**Comment by JordanMartinez** [comment]
Found it! [This line](https://github.com/FXMisc/RichTextFX/blob/master/richtextfx/src/main/java/org/fxmisc/richtext/StyledTextArea.java#L97) adds the `text` style class to a text.

**Comment by mat4pat** [comment]
Hello, I see that this problem doesn't have a clear solution.
I have read the CSS Ref Guide, but there is no "efficient" solution to @AleKrabbe's question:
How do you give a default color to all **non-syntax-highlighted** text?

I have tried @AleKrabbe's solution: add !important property after each keyword '-fx-fill', as so:
![image](https://user-images.githubusercontent.com/85194618/150652253-28a94e2f-b201-4cc0-9419-dc4bb0f35810.png)

And then color-filling all the text to my preferred color:
![image](https://user-images.githubusercontent.com/85194618/150652280-56176434-b137-4df1-977c-10704749bfb8.png)
This way every color in my CodeArea instance is white, EXCEPT all the keywords with the !important property.

As you can see, the color of all the non-syntax-highlighted text is infact the color I wanted(#9affc0 - whitish color):
![image](https://user-images.githubusercontent.com/85194618/150652422-a5498e94-ead1-4542-ad71-e93df574fefe.png)
But as you may notice, the color of lines is also the same color.

So it practically ignores the CSS property of the "lineno" class, the rectangle that holds all the lines of the CodeArea.
I'm taking about this line:
![image](https://user-images.githubusercontent.com/85194618/150652534-6651cb33-261f-4b49-a223-cab354b0d15d.png)
This will not change to color of the lines to my preferred color, even if I add the !important property in line 64.

So TL:DR - Is there any easy way to color all the non-sytnax-highlighted text in a CodeArea, without altering the color of lines etc.
This is practically a similar question to ISSUE #442  - https://github.com/FXMisc/RichTextFX/issues/442.

Thank you very much, I really appreciate the hard work on this library, keep it up!

**Comment by Jugen** [comment]
Try changing your CSS tag `#codearea .text` to `#codearea .paragraph-text .text`

**Comment by mat4pat** [comment]
@Jugen Thank you very much, it works just as I want it to:
![image](https://user-images.githubusercontent.com/85194618/150680810-1a98348f-d40e-47a6-ae04-10549a9a3330.png)

For anyone else reading in the future, don't forget to add the !important property, as I did above.
